### PR TITLE
Check for implicitly unwrapped relationships in `getOptionalProperties`

### PR DESCRIPTION
**Pull request by Anviking**
Quick, potential fix for #3768

Playing around in a playground I found that `childMirror.displayStyle` for a `T!` (where `T: Object`) is `.enum` while `childMirror.displayStyle` for `T?` is `.optional` using Swift 3.0. Haven't explicitly tested on any previous Swift version, but I assume this is what's causing the problem.

Also haven't verified that the specific implicitly unwrapped properties in my app works, but it does at least launch now.

### CODE REVIEW

**Comment by jpsim** [comment]
Hi @Anviking, thanks for doing this (and accepting our CLA, I just verified that). This would bring us back to the behavior in Swift 2.2, which I'm in favor of. Realm can never guarantee that an object property will _always_ be fulfilled, but if the user (e.g. you) explicitly ignores that, we should continue to support that behavior IMHO.

We also had unit tests that verified this behavior in Swift 2.2 ([source](https://github.com/realm/realm-cocoa/blob/9c978d96fc3e6cbdf446a0e2b28591434e52daa3/RealmSwift-swift2.2/Tests/ObjectSchemaInitializationTests.swift#L190-L197)). Could you also re-introduce these in the Swift 3 tests please?

**Comment by Anviking** [comment]
@jpsim done!

**Comment by Anviking** [comment]
(And rebased against master yesterday)

File: RealmSwift/Tests/ObjectSchemaInitializationTests.swift [hunk 1]
```diff
@@ -182,6 +182,17 @@ class ObjectSchemaInitializationTests: TestCase {
         let types = Set(schema.properties.map { $0.type })
         XCTAssertEqual(types, Set([.string, .string, .data, .date, .object, .int, .float, .double, .bool]))
     }
+    
+    func testImplicitlyUnwrappedOptionalProperties() {
```
Comment (jpsim) [inline RealmSwift/Tests/ObjectSchemaInitializationTests.swift]: Is there a reason the Swift 2.2 test wasn't sufficient? It's important to keep differences between Swift version implementations to a minimum to facilitate the transition.

```swift
func testImplicitlyUnwrappedOptionalsAreParsedAsOptionals() {
    let schema = SwiftImplicitlyUnwrappedOptionalObject().objectSchema
    XCTAssertTrue(schema["optObjectCol"]!.optional)
    XCTAssertTrue(schema["optNSStringCol"]!.optional)
    XCTAssertTrue(schema["optStringCol"]!.optional)
    XCTAssertTrue(schema["optBinaryCol"]!.optional)
    XCTAssertTrue(schema["optDateCol"]!.optional)
}
```

**Comment by jpsim** [comment]
Thanks for updating this @Anviking! Just had one last comment about the unit test.

File: RealmSwift/Tests/ObjectSchemaInitializationTests.swift (hunk 1 above)
Comment (Anviking) [inline RealmSwift/Tests/ObjectSchemaInitializationTests.swift]: Ok! Not at all, just falsely believed style in `testOptionalProperties` had been introduced after this PR when I rebased.

Comment (Anviking) [inline RealmSwift/Tests/ObjectSchemaInitializationTests.swift]: Fixed and squashed now!

**Comment by jpsim** [comment]
Looks good. Thanks for your contribution @Anviking!

**Pull request merged**